src/Compiler/Microsoft.CodeAnalysis.Razor/src/SymbolExtensions.cs

```diff
@@ -14,6 +14,23 @@ internal static class SymbolExtensions
 
     internal static bool HasFullName(this ISymbol symbol, string fullName)
     {
+        // generally we expect this call to fail, so try and match on the simple name first, before
```

Comment:
If I recall from previous PR, we use this API on specific full names, which have maybe 2 or 3 parts (namespace-namespace-name). 
I don't remember if there were generic names passed it.
If not, consider making `HasFullName(this ISymbol symbol, string containing1, string name)` and `HasFullName(this ISymbol symbol, string containing1, string containing2, string name)` APIs instead.

Those would just do direct name comparisons (and verify zero arity) and then we would never need to do `.ToDisplayString`.

Reply:
We do have generic types unfortunately. I can play around with not using ToDisplayString and see what it looks like.

Reply:
From offline discussion, this PR does solve a problem even if there's opportunity to go further. I'm okay if we do a follow-up to do more optimization.